File: Household Budget Calculator/[C#]-Household Budget Calculator/C#/Household budget calculator CS/expanding property classes/Children.cs
//using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.ComponentModel;


namespace Household_budget_calculator_CS
{
    [TypeConverter(typeof(ExpandableObjectConverter))]
    [System.Serializable()]
    public class Children
    {
        public event propertiesChangedEventHandler propertiesChanged;
        public delegate void propertiesChangedEventHandler(propertiesChangedEventArgs e);

        private decimal _care;
        [DisplayName("Daycare/Nanny"), RefreshProperties(RefreshProperties.All)]
        public decimal care
        {
            get { return _care; }
            set
            {
                _care = value;
                totalChildren = care + schooling + clothing + treats;
            }
        }

        private decimal _schooling;
        [DisplayName("Schooling"), RefreshProperties(RefreshProperties.All)]
        public decimal schooling
        {
            get { return _schooling; }
            set
            {
                _schooling = value;
                totalChildren = care + schooling + clothing + treats;
            }
        }

        private decimal _clothing;
        [DisplayName("Clothing"), RefreshProperties(RefreshProperties.All)]
        public decimal clothing
        {
            get { return _clothing; }
            set
            {
                _clothing = value;
                totalChildren = care + schooling + clothing + treats;
            }
        }

        private decimal _treats;
        [DisplayName("Treats"), RefreshProperties(RefreshProperties.All)]
        public decimal treats
        {
            get { return _treats; }
            set
            {
                _treats = value;
                totalChildren = care + schooling + clothing + treats;
            }
        }

        private decimal _totalChildren;
        [ReadOnlyAttribute(true), Browsable(false), DisplayName("Total Children Outgoings")]
        public decimal totalChildren
        {
            get { return _totalChildren; }
            set
            {
                _totalChildren = value;
                if (propertiesChanged != null)
                {
                    propertiesChanged(new propertiesChangedEventArgs
                    {
                        propName = "Children",
                        newValue = value
                    });
                }
            }
        }

        public override string ToString()
        {
            return totalChildren.ToString("c2");
        }

    }
}

File: Household Budget Calculator/[VB]-Household Budget Calculator/VB/Household budget calculator/expanding property classes/Children.vb
Imports System.ComponentModel

<TypeConverter(GetType(ExpandableObjectConverter))> <System.Serializable()> _
Public Class Children

    Public Event propertiesChanged(ByVal e As propertiesChangedEventArgs)

    Private _care As Decimal
    <DisplayName("Daycare/Nanny"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property care() As Decimal
        Get
            Return _care
        End Get
        Set(ByVal value As Decimal)
            _care = value
            totalChildren = care + schooling + clothing + treats
        End Set
    End Property

    Private _schooling As Decimal
    <DisplayName("Schooling"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property schooling() As Decimal
        Get
            Return _schooling
        End Get
        Set(ByVal value As Decimal)
            _schooling = value
            totalChildren = care + schooling + clothing + treats
        End Set
    End Property

    Private _clothing As Decimal
    <DisplayName("Clothing"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property clothing() As Decimal
        Get
            Return _clothing
        End Get
        Set(ByVal value As Decimal)
            _clothing = value
            totalChildren = care + schooling + clothing + treats
        End Set
    End Property

    Private _treats As Decimal
    <DisplayName("Treats"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property treats() As Decimal
        Get
            Return _treats
        End Get
        Set(ByVal value As Decimal)
            _treats = value
            totalChildren = care + schooling + clothing + treats
        End Set
    End Property

    Private _totalChildren As Decimal
    <ReadOnlyAttribute(True), _
    Browsable(False), _
    DisplayName("Total Children Outgoings")> _
    Public Property totalChildren() As Decimal
        Get
            Return _totalChildren
        End Get
        Set(ByVal value As Decimal)
            _totalChildren = value
            RaiseEvent propertiesChanged(New propertiesChangedEventArgs With {.propName = "Children", .newValue = value})
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return totalChildren.ToString("c2")
    End Function

End Class
